server/blob/blob-joining/src/main/java/org/apache/james/blob/joining/JoiningBlobStore.java

```diff
@@ -26,52 +26,133 @@
 
 import org.apache.james.blob.api.BlobId;
 import org.apache.james.blob.api.BlobStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.github.fge.lambdas.Throwing;
 import com.google.common.annotations.VisibleForTesting;
 
 public class JoiningBlobStore implements BlobStore {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(JoiningBlobStore.class);
+
+    private static class JoiningBlobStoreFallBack implements BlobStore {
```

Comment:
Why do you need this private internal blob store?

I would bet that method extraction will look easier to read.